src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs

```diff
@@ -0,0 +1,1482 @@
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+
+namespace Okta.Sdk.IntegrationTests
+{
+    [Collection(nameof(PoliciesScenarios))]
+    public class PoliciesScenarios
+    {
+        [Fact]
+        public async Task CreateSignOnPolicy()
+        {
+            var client = TestClient.Create();
+
+            IPolicy policy = new Policy()
+            {
+                // Name has a maximum of 50 chars
+                Name = $"Default policy {Guid.NewGuid()}".Substring(0, 50),
+                Type = PolicyType.OktaSignOn,
+                Status = "ACTIVE",
+                Description = "The default policy applies in all situations if no other policy applies.",
+            };
+
+            var createdPolicy = await client.Policies.CreatePolicyAsync(policy);
+
+            try
+            {
+                createdPolicy.Should().NotBeNull();
+                createdPolicy.Name.Should().Be(policy.Name);
+                createdPolicy.Type.Should().Be(PolicyType.OktaSignOn);
+                createdPolicy.Status.Should().Be("ACTIVE");
+                createdPolicy.Description.Should().Be(policy.Description);
+            }
+            finally
+            {
+                await client.Policies.DeactivatePolicyAsync(createdPolicy.Id);
+                await client.Policies.DeletePolicyAsync(createdPolicy.Id);
+            }
+        }
+
+        [Fact]
+        public async Task GetAPolicy()
+        {
+            var client = TestClient.Create();
+
+            IPolicy policy = new Policy()
+            {
+                // Name has a maximum of 50 chars
+                Name = $"Default policy {Guid.NewGuid()}".Substring(0, 50),
+                Type = PolicyType.OktaSignOn,
+                Status = "ACTIVE",
+                Description = "The default policy applies in all situations if no other policy applies.",
+            };
+
+            var createdPolicy = await client.Policies.CreatePolicyAsync(policy);
+
+            try
+            {
+                var retrievedPolicy = await client.Policies.GetPolicyAsync(createdPolicy.Id);
+
+                retrievedPolicy.Should().NotBeNull();
+                retrievedPolicy.Name.Should().Be(policy.Name);
+                retrievedPolicy.Type.Should().Be(PolicyType.OktaSignOn);
+                retrievedPolicy.Status.Should().Be("ACTIVE");
+                retrievedPolicy.Description.Should().Be(policy.Description);
+            }
+            finally
+            {
+                await client.Policies.DeactivatePolicyAsync(createdPolicy.Id);
+                await client.Policies.DeletePolicyAsync(createdPolicy.Id);
+            }
+        }
+
+        [Fact]
+        public async Task GetAPolicyOfType()
```

Comment:
I see that this test is adding a group condition and the next test, `CreateSignOnPolicyWithGroupConditions`, is also doing the same thing?  Do we need to do this twice?  It seems like `CreateSignOnPolicyWithGroupConditions` could keep the group conditions (a good test) but `GetAPolicyOfType` could drop that, and create a basic sign-on policy and then query for it.  I'm also wondering about the name of this test, `GetAPolicyOfType`?  It seems like it's just getting a single policy by ID, not querying for all policies of a given type?

Reply:
@robertjd agree about creating a basic sign on policy. 
In `GetAPolicyOfType` the difference with a simple `get` method, is that we are passing the `type` of the object to be retrieved, in this case `IOktaSignOnPolicy`. 
`var retrievedPolicy = await client.Policies.GetPolicyAsync<IOktaSignOnPolicy>(createdPolicy.Id);`

Reply:
Ah I see, so this test is asserting that the SDK creates the right type?  If so that makes sense.